### PR TITLE
Groups members frontend

### PR DESCRIPTION
**Pull request by hamzashariq7**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,6 +32,8 @@ gem "paperclip", "~> 6.0.0"
 
 gem 'devise'
 
+gem 'chosen-rails'
```
Comment (zainsaleem4338) [inline Gemfile:35]: add version '1.5.2'
https://rubygems.org/gems/chosen-rails/versions/1.5.2

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -20,7 +20,7 @@ class ApplicationController < ActionController::Base
     unless request.subdomain.empty?
       @current_company = Company.find_by_subdomain(request.subdomain)
 
-      if @current_company == nil
+      unless @current_company.present?
```
Comment (zohaibbutt13) [inline app/controllers/application_controller.rb]: convert to if

File: app/controllers/groups_controller.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ def create
         flash[:notice] = "Group created successfully!"
         format.html { redirect_to groups_path }
       else
+        flash[:error] = "#{ @group.errors.messages.first[0] } #{ @group.errors.messages.first[1][0] }"
```
Comment (zohaibbutt13) [inline app/controllers/groups_controller.rb]: why need this ? "#{ @group.errors.messages.first[0] } #{ @group.errors.messages.first[1][0] }"
can we use @group.errors.full_messages ?

File: app/controllers/groups_controller.rb [hunk 4]
```diff
@@ -25,10 +26,11 @@ def edit
 
   def update
     respond_to do |format|
-      if @group.update_attributes!(group_params)
+      if @group.update_attributes(group_params)
         flash[:notice] = "Group updated successfully!"
         format.html { redirect_to groups_path }
```
Comment (zohaibbutt13) [inline app/controllers/groups_controller.rb]: redirect to detail page (throughout the app)

File: app/models/ability.rb [hunk 5]
```diff
@@ -7,7 +7,7 @@ def initialize(user)
       can :manage, [User, Group, Survey], company_id: user.company_id
       can :manage, UserSetting, user_id: user.id
       can :manage, CompanySetting, company_id: user.company_id
-      cannot [:destroy, :edit], user
+      cannot [:destroy, :edit_in_members_view], user
```
Comment (zohaibbutt13) [inline app/models/ability.rb]: why not just edit ?

File: app/models/group.rb [hunk 6]
```diff
@@ -5,12 +5,17 @@ class Group < ActiveRecord::Base
   belongs_to :company
   has_and_belongs_to_many :users
 
-  validates :name, presence: true, length: { maximum: 150 }
-  validates :description, presence: true, length: { maximum: 500 }
+  validates :name,
```
Comment (zohaibbutt13) [inline app/models/group.rb]: single line

File: app/views/groups/_fields.js.erb [hunk 7]
```diff
@@ -0,0 +1 @@
+$('.tokenfield').tokenfield()
```
Comment (zohaibbutt13) [inline app/views/groups/_fields.js.erb]: always add ; in js code

File: app/views/groups/index.html.erb [hunk 8]
```diff
@@ -1,22 +1,34 @@
...
       <tr>
+        <th scope="row"><%= index %></th>
         <td><%= group.name %></td>
-        <% if can? :manage, group %>
-          <td><%= link_to 'Edit', edit_group_path(group) %></td>
-          <td><%= link_to 'Delete', group, method: :delete, data: { confirm: 'Are you sure?' } %></td>
-        <% end %>
+        <td><%= group.description %></td>
+        <td>
+          <% if can? :manage, group %>
```
Comment (zohaibbutt13) [inline app/views/groups/index.html.erb]: what if user can only edit group but not destroy ?

File: app/controllers/groups_controller.rb [hunk 9]
```diff
@@ -5,6 +5,9 @@ class GroupsController < ApplicationController
     @employees = User.all
   end
 
+  def show
```
Comment (zohaibbutt13) [inline app/controllers/groups_controller.rb:8]: respond_to block in every action

File: app/models/user.rb [hunk 10]
```diff
@@ -28,22 +27,26 @@ class User < ActiveRecord::Base
 
   accepts_nested_attributes_for :company
 
-  validates :first_name, presence: true, length: { maximum: 150 }
-  validates :last_name, presence: true, length: { maximum: 150 }
+  validates :first_name,
```
Comment (zohaibbutt13) [inline app/models/user.rb]: single line

File: app/views/groups/_fields.js.erb (hunk 7 above)
Comment (zohaibbutt13) [inline app/views/groups/_fields.js.erb]: ;

File: app/assets/javascripts/member_token_input.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+$(document).ready(function () {
+  $("#group_user_ids").chosen({
+    placeholder_text_multiple: "Select Some Members"
```
Comment (zohaibbutt13) [inline app/assets/javascripts/member_token_input.js]: add ;

**Pull request merged**